common/src/main/scala/org/apache/celeborn/common/meta/WorkerInfo.scala

```diff
@@ -188,11 +188,14 @@ class WorkerInfo(
     JavaUtils.newConcurrentHashMap[String, DiskInfo](diskInfos)
   }
 
-  def updateThenGetUserResourceConsumption(consumption: util.Map[
+  def updateThenGetUserResourceConsumption(resourceConsumptions: util.Map[
     UserIdentifier,
     ResourceConsumption]): util.Map[UserIdentifier, ResourceConsumption] = {
-    userResourceConsumption.clear()
-    userResourceConsumption.putAll(consumption)
+    userResourceConsumption.keys().asScala.filterNot(resourceConsumptions.containsKey).foreach {
```

Comment:
I think this change is incorrect.  According to this logic, If an identifier is gone, there will always be zero resource consumption attached to it.
I think make the "clear" and "putall" in a atomic block should be ok instead of this change.

Reply:
I think just removing the inexistent idenfiers from user resource consumption will be better.